File: counting_elements/count_elements.go
package main

/**
 * Time: O(n)
 * Space: O(n)
 */
func countElements(arr []int) int {
	s := make(map[int]bool)
	for _, i := range arr {
		s[i] = true
	}
	count := 0
	for _, i := range arr {
		if s[i+1] == true {
			count++
		}
	}
	return count
}

func main() {

}

File: counting_elements/count_elements.cpp
#include <iostream>
#include <unordered_set>
#include <vector>


/**
 * Time: O(n) 
 * Space: O(n)
 */
int count_elements(vector<int> &arr)
{
    unordered_set<int> s{arr.begin(), arr.end()};
    int count = 0;
    for (auto i : arr)
        if (s.count(i + 1))
            count++;
    return count;
}